### PR TITLE
feat: Subject Segmentation

### PR DESCRIPTION
**Pull request by bensonarafat**
Subject Segmentation based on issue #586 

ML Kit's subject segmentation API allows developers to easily separate multiple subjects from the background in a picture, enabling use cases such as sticker creation, background swap, or adding cool effects to subjects. [Read more](https://developers.google.com/ml-kit/vision/subject-segmentation) 

Key capabilities
- Multi-subject segmentation: provides masks and bitmaps for each individual subject, rather than a single mask and bitmap for all subjects combined.
- Subject recognition: subjects recognized are objects, pets, and humans.
- On-device processing: all processing is performed on the device, preserving user privacy and requiring no network connectivity.

**This API requires Android API level 24 or above. Make sure that your app's build file uses a minSdkVersion value of 24 or higher**

### CODE REVIEW

**Comment by fbernaly** [comment]
@bensonarafat : fix the static code analysis

File: packages/google_mlkit_subject_segmentation/ios/Classes/GoogleMlkitSubjectionSegmentationPlugin.m [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
++ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
+  FlutterMethodChannel* channel = [FlutterMethodChannel
+      methodChannelWithName:@"google_mlkit_subject_segmentation"
+            binaryMessenger:[registrar messenger]];
+  GoogleMlKitSubjectSegmentationPlugin* instance = [[GoogleMlKitSubjectSegmentationPlugin alloc] init];
+  [registrar addMethodCallDelegate:instance channel:channel];
+}
+
+- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
+  if ([@"getPlatformVersion" isEqualToString:call.method]) {
```
Comment (fbernaly) [inline packages/google_mlkit_subject_segmentation/ios/Classes/GoogleMlkitSubjectionSegmentationPlugin.m:13]: Delete this `if`, just return:
result(FlutterMethodNotImplemented);

**Comment by fbernaly** [comment]
@bensonarafat : These are the errors you need to fix:

Screenshot 2024-08-12 at 9 38 53 AM

**Comment by bensonarafat** [comment]
@fbernaly 
Yes.
I forgot about the code-analysis.yaml file.

**Comment by bensonarafat** [comment]
@fbernaly 
We are good

**Pull request merged**